test/unit/auxiliary/mc/test_norm.cpp

```diff
@@ -0,0 +1,164 @@
+
+// Given a global index of an element, change its value using setter
+// A reference to the elemenet is given as parameter to the setter function [void (*setter)(T& element)]
+template <class T, class ElementSetter>
+void modify_element(Matrix<T, Device::CPU>& matrix, GlobalElementIndex index, ElementSetter set) {
+  const auto& distribution = matrix.distribution();
+
+  // clang-format off
+  GlobalTileIndex tile_index{
+    distribution.template globalTileFromGlobalElement<Coord::Row>(index.row()),
+    distribution.template globalTileFromGlobalElement<Coord::Col>(index.col())
+  };
+  // clang-format on
+
+  if (distribution.rankIndex() != distribution.rankGlobalTile(tile_index))
+    return;
+
+  // clang-format off
+  const TileElementIndex index_wrt_local{
+    distribution.template tileElementFromGlobalElement<Coord::Row>(index.row()),
+    distribution.template tileElementFromGlobalElement<Coord::Col>(index.col())
+  };
+  // clang-format on
+
+  matrix(tile_index).then(hpx::util::unwrapping([set, index_wrt_local](auto&& tile) {
+    set(tile(index_wrt_local));
+  }));
+}
+
+// Change the specified value of the matrix, re-compute the norm with given parameters and check if the
+// result is the expected one
+template <class T>
+void set_and_test(CommunicatorGrid comm_grid, Matrix<T, Device::CPU>& matrix, GlobalElementIndex index,
+                  T new_value, NormT<T> norm_expected, lapack::Norm norm_type, blas::Uplo uplo) {
+  if (index.isIn(matrix.size()))
+    modify_element(matrix, index, [new_value](T& element) { element = new_value; });
+
+  const NormT<T> norm = Auxiliary<Backend::MC>::norm(comm_grid, norm_type, uplo, matrix);
+
+  SCOPED_TRACE(::testing::Message() << lapack::norm2str(norm_type) << " " << blas::uplo2str(uplo)
+                                    << " changed element=" << index << " in matrix size="
+                                    << matrix.size() << " grid_size=" << comm_grid.size());
+
+  if (Index2D{0, 0} == comm_grid.rank())
+    EXPECT_NEAR(norm_expected, norm, TypeUtilities<NormT<T>>::error);
+}
+
+TYPED_TEST(NormDistributedTest, NormMax) {
+  const lapack::Norm norm_type = lapack::Norm::Max;
+
+  const std::vector<GlobalElementSize> sizes({{10, 10}, {0, 0}});
+  const std::vector<TileElementSize> block_sizes({{3, 3}, {5, 5}});
+
+  for (const auto& comm_grid : this->commGrids()) {
+    for (const auto& size : sizes) {
+      for (const auto& block_size : block_sizes) {
+        Index2D src_rank_index(std::max(0, comm_grid.size().rows() - 1),
+                               std::min(1, comm_grid.size().cols() - 1));
+        Distribution distribution(size, block_size, comm_grid.size(), comm_grid.rank(), src_rank_index);
+        Matrix<TypeParam, Device::CPU> matrix(std::move(distribution));
+
+        for (const auto& uplo : blas_uplos) {
+          if (blas::Uplo::Lower != uplo)
+            continue;
+
+          dlaf::matrix::util::set_random(matrix);
```

Comment:
for uplo == Lower/Upper the matrix should be either symmetric or hermitian, (i.e. for complex matrices all the diagonal elements should be either real, or purely imaginary)

Reply:
In order to address this I've just opened #237, waiting for that before fixing this.

Reply:
#237 has been embedded in this PR, and now that line has been changed to `set_random_hermitian`.